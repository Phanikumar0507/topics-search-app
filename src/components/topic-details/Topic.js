import React from "react";

const Topic = ({ topic, setShowRelated }) => {
  return (
    <div className="topic" onClick={() => setShowRelated(true)}>
      <div className="topic-name">{topic.name}</div>
      <div className="topic-star-count">{topic.stargazerCount}</div>
    </div>
  );
};

export default Topic;
