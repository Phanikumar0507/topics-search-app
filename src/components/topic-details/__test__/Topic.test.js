import Topic from "../Topic";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

describe("Topic Component", () => {
  it("should render without crash", () => {
    const component = shallow(
      <Topic topic={{ name: "react", stargazerCount: 100 }} />
    );
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("should print the passed props in the name and count", () => {
    const component = shallow(
      <Topic topic={{ name: "angular", stargazerCount: 999 }} />
    );
    expect(component.find(".topic-name").text()).toEqual("angular");
    expect(component.find(".topic-star-count").text()).toEqual("999");
  });

  it("should call the callback on topic click", () => {
    const topic = { name: "vue", stargazerCount: 999 };
    const mockFunc = jest.fn();
    const component = shallow(
      <Topic topic={topic} setShowRelated={mockFunc} />
    );
    component.find(".topic").simulate("click");
    expect(mockFunc).toHaveBeenCalledTimes(1);
    expect(mockFunc).toHaveBeenCalledWith(true);
  });
});
