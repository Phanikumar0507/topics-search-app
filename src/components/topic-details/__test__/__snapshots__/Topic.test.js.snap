// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Component should render without crash 1`] = `
<div
  className="topic"
  onClick={[Function]}
>
  <div
    className="topic-name"
  >
    react
  </div>
  <div
    className="topic-star-count"
  >
    100
  </div>
</div>
`;
